File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
  const answer = [];

  for (const item of targets) {
    let count = 0;
    for (const s of item) {
      let minIndex = Infinity;
      for (const key of keymap) {
        if (key.includes(s)) {
          const index = key.indexOf(s);
          minIndex = Math.min(minIndex, index);
        }
      }
      if (minIndex !== Infinity) count += minIndex + 1;
      else {
        count = 0;
        break;
      }
    }
    count ? answer.push(count) : answer.push(-1);
  }

  return answer;
}

File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    answer = []
    for target in targets:
        sumValue = 0
        for string in target:
            minValue = 1000
            for key in keymap:
                if key.find(string) != -1:
                    minValue = min(minValue, key.find(string) + 1)
            sumValue += minValue
        if sumValue >= 1000:
            answer.append(-1)
        else:
            answer.append(sumValue)
    return answer